firedrake/interpolation.py

```diff
@@ -46,261 +46,142 @@
 )
 
 
+@dataclass
+class InterpolateOptions:
+    """Options for interpolation operations.
+
+    Attributes
+    ----------
+    subset : pyop2.types.set.Subset, optional
+        An optional subset to apply the interpolation over.
+        Cannot, at present, be used when interpolating across meshes unless
+        the target mesh is a :func:`.VertexOnlyMesh`.
+    access : pyop2.types.access.Access, default op2.WRITE
+        The pyop2 access descriptor for combining updates to shared
+        DoFs. Possible values include ``WRITE`` and ``INC``. Only ``WRITE`` is
+        supported at present when interpolating across meshes unless the target
+        mesh is a :func:`.VertexOnlyMesh`.
+
+        .. note::
```

Comment:
The way we handle INC now always zeroes the ouput, so this is out of date. We might want another kwarg `needs_zeroing` like the one we have for `assemble`, the default is True.